index.bs

```diff
@@ -2289,10 +2297,33 @@ during registration.
 </div>
 
+#### Easily accessing credential data #### {#sctn-public-key-easy}
+
+Every user of the [[#sctn-createCredential|create method]] will need to parse and store the returned public key in order to verify future assertions. However, the public key is in [[!RFC8152]] (COSE) format, inside the [=credentialPublicKey=] member of the [=attestedCredentialData=], inside the [=authenticator data=], inside the [=attestation object=] from {{AuthenticatorAttestationResponse/attestationObject}}. Users that wish to use [=attestation=] MUST do the work of parsing that out because that copy of the public-key is the one that the [=authenticator=] has [signed](#signing-procedure). However, many valid uses of WebAuthn do not need [=attestation=]. For those uses, user agents can do the work of parsing, expose the [=authenticator data=] directly, and translate the public key into a more convenient format.
+
+The {{AuthenticatorAttestationResponse/getPublicKey()}} operation thus returns the credential public key as a [=SubjectPublicKeyInfo=]. After base64 decoding, this format can, for example, be passed to Java's `java.security.spec.X509EncodedKeySpec`, .NET's `System.Security.Cryptography.ECDsa.ImportSubjectPublicKeyInfo`, or Go's `crypto/x509.ParsePKIXPublicKey`.
+
+Use of this function does impose some limitations: by using {{PublicKeyCredentialCreationOptions/pubKeyCredParams}}, a [=[RP]=] can negotiate with the [=authenticator=] to use public-key algorithms that the user agent may not understand. However, if it does so, the user agent will not be able translate the resulting public key into [=SubjectPublicKeyInfo=] format and the return value of {{AuthenticatorAttestationResponse/getPublicKey()}} will be an empty string.
+
+User agents MUST be able to provide a non-empty value for {{AuthenticatorAttestationResponse/getPublicKey()}} when the [=credential public key=] has a {{COSEAlgorithmIdentifier}} value of:
+   * -7 (ECDSA with SHA-256), where [=kty=] is 2 (uncompressed points) and [=crv=] is 1 (P-256).
+   * -257 (RSA PKCS#1 with SHA-256).
+   * -8 (EdDSA), where [=crv=] is 6 (Ed25519).
+
+To remove the need to parse CBOR completely in many cases, {{AuthenticatorAttestationResponse/getAuthenticatorData()}} returns the [=authenticator data=] from {{AuthenticatorAttestationResponse/attestationObject}}. The [=authenticator data=] contains other fields that are encoded in a binary format. However, helper functions are not provided to access them because [=[RPS]=] already need to extract those fields when [getting an assertion](#sctn-getAssertion). In contrast to [credential creation](#sctn-createCredential), where signature verification is [optional](#enumdef-attestationconveyancepreference), [=[RPS]=] should always be verifying signatures from an assertion and thus must extract fields from the signed [=authenticator data=]. The same functions used there will also serve during credential creation.
```

Comment:
Maybe?
(Change "parse CBOR completely" to "parse CBOR at all")

```suggestion
To remove the need to parse CBOR at all in many cases, {{AuthenticatorAttestationResponse/getAuthenticatorData()}} returns the [=authenticator data=] from {{AuthenticatorAttestationResponse/attestationObject}}. The [=authenticator data=] contains other fields that are encoded in a binary format. However, helper functions are not provided to access them because [=[RPS]=] already need to extract those fields when [getting an assertion](#sctn-getAssertion). In contrast to [credential creation](#sctn-createCredential), where signature verification is [optional](#enumdef-attestationconveyancepreference), [=[RPS]=] should always be verifying signatures from an assertion and thus must extract fields from the signed [=authenticator data=]. The same functions used there will also serve during credential creation.
```